lib/automaticupgrades/version.go

```diff
@@ -73,3 +72,60 @@ func Version(ctx context.Context, baseURL string) (string, error) {
 
 	return versionString, trace.Wrap(err)
 }
+
+// getVersionURL returns the versionURL or the default stable/cloud version url.
+func getVersionURL(versionURL string) (string, error) {
+	if versionURL != "" {
+		return versionURL, nil
+	}
+	cloudStableVersionURL, err := url.JoinPath(stableCloudVersionBaseURL, stableCloudVersionPath)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	return cloudStableVersionURL, nil
+}
+
+// Critical returns true if a critical upgrade is available.
+func Critical(ctx context.Context, criticalURL string) (bool, error) {
+	criticalURL, err := getCriticalURL(criticalURL)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, criticalURL, nil)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	defer resp.Body.Close()
+
+	body, err := utils.ReadAtMost(resp.Body, teleport.MaxHTTPResponseSize)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return false, trace.BadParameter("invalid status code %d, body: %s", resp.StatusCode, string(body))
+	}
+
+	critical := strings.TrimSpace(string(body))
+
+	// The critical endpoint returns either the string "yes" or "no"
+	return critical == "yes", nil
```

Comment:
The body can have different payload can we handled these value separate and returns a error if  value is different that yes or no ?